app/__init__.py

```diff
@@ -0,0 +1,48 @@
+import os
+
+from flask import Flask
+from flask_caching import Cache
+from flask_compress import Compress
+from app.dalton import dalton_blueprint, ensure_rulesets_exist, setup_dalton_logging
+from app.flowsynth import flowsynth_blueprint, setup_flowsynth_logging
+import logging
+
+
+def create_app(test_config=None):
+    """Create the flask app."""
+    curdir = os.path.dirname(os.path.abspath(__file__))
+    static_folder = os.path.join(curdir, "static")
+    daltonfs = Flask("app", static_folder=static_folder)
+    if test_config:
+        # load the test config if passed in
+        daltonfs.config.from_mapping(test_config)
+
+    if not daltonfs.testing:
+        setup_dalton_logging()
+        setup_flowsynth_logging()
+        ensure_rulesets_exist()
+
+    # register modules
+    #
+    # dalton
+    daltonfs.register_blueprint(dalton_blueprint)
```

Comment:
This code to create the app was copied from `run.py`... then tweaked a little so that we could run the tests.

Only when not running the tests, do we need to call the new functions
- setup_*_logging ; and
- ensure_rulesets_exist()

... since otherwise, that code was run at import time.